### PR TITLE
Miscellaneous updates and cleanup

### PR DESCRIPTION
**Pull request by TacoTheDank**
# Important

Please confirm that you:

[x] read the [contributing section](https://github.com/M66B/FairEmail#user-content-contributing)
[x] agree to [the license and the copyright](https://github.com/M66B/FairEmail#user-content-license)

Thanks for your intention to contribute to the project!

### 1st commit
- Add gradle wrapper validation action (see [this](https://github.com/gradle/wrapper-validation-action))

### 2nd commit
- Clean up .gitignore
- Update minimum cmake version
- Remove unused lifecycle-extensions library

### 3rd commit
- Delete generated .idea

### 4th commit
- Fix a few performance lints

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -12,6 +12,8 @@ keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
 
 android {
     compileSdkVersion 30
+    buildToolsVersion "30.0.3"
+    ndkVersion "22.0.7026061"
```
Comment (M66B) [inline app/build.gradle:16]: For reproducible builds this would be nice, but for the rest it is not really useful.

**Comment by M66B** [comment]
JavaMail and OpenPGP API should not be changed when there is no need because this will make synchronizing with the original source code more difficult.

**Comment by M66B** [comment]
If I don't merge everything, this doesn't mean I don't appreciate what you are trying to do here!

File: app/build.gradle (hunk 1 above)
Comment (TacoTheDank) [reply app/build.gradle:16]: 🤔 I see

**Comment by TacoTheDank** [comment]
> If I don't merge everything, this doesn't mean I don't appreciate what you are trying to do here!

It's completely fine! Thanks 😃

**Review by M66B — COMMENTED** [review]
What I wanted to say is that each time the build tools are updated or the NDK is updated, even if it is a minor version update, the gradle file needs to be changed.

File: app/build.gradle [hunk 2]
```diff
@@ -329,9 +328,6 @@ dependencies {
     implementation "androidx.lifecycle:lifecycle-livedata-core:$lifecycle_version"
     annotationProcessor "androidx.lifecycle:lifecycle-compiler:$lifecycle_version"
 
-    // https://mvnrepository.com/artifact/androidx.lifecycle/lifecycle-extensions
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/build.gradle:330]: It was not used anywhere in the project.

File: app/src/main/java/eu/faircode/email/ActivityView.java [hunk 3]
```diff
@@ -1122,8 +1122,7 @@ private void onMenuOutbox() {
             @Override
             protected EntityFolder onExecute(Context context, Bundle args) {
                 DB db = DB.getInstance(context);
-                EntityFolder outbox = db.folder().getOutbox();
-                return outbox;
+                return db.folder().getOutbox();
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/src/main/java/eu/faircode/email/ActivityView.java:1125]: I see 🤔

File: .github/workflows/gradle-wrapper-validation.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+name: "Validate Gradle Wrapper"
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply .github/workflows/gradle-wrapper-validation.yml:1]: It's actually fine, no external dependency is used. [Slide](https://github.com/ccrama/Slide) uses this validation tool and still builds with F-Droid.

File: app/CMakeLists.txt [hunk 5]
```diff
@@ -1,7 +1,7 @@
 
 # https://developer.android.com/studio/projects/configure-cmake
 
-cmake_minimum_required(VERSION 3.4.1)
+cmake_minimum_required(VERSION 3.10.2)
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/CMakeLists.txt:4]: In what manner, exactly? I built an APK and tested it, and it seemed to work perfectly fine. Is there another way it should be tested? I'm not very familiar with CMake stuff, really.

File: .gitignore [hunk 6]
```diff
@@ -1,27 +1,16 @@
 *.iml
 .gradle
-/local.properties
-/.idea/caches
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply .gitignore:4]: Are you saying you want to keep some of the .idea files?

File: .gitignore [hunk 7]
```diff
@@ -1,27 +1,16 @@
 *.iml
 .gradle
-/local.properties
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply .gitignore:3]: I mean, it works fine in this case. I've had a few times where weird stuff like not-ignored files happen, and this seems to solve it.

**Comment by M66B** [comment]
Lifecycle extensions are definitely required (in SimpleTask.java).

**Comment by TacoTheDank** [comment]
> Lifecycle extensions are definitely required (in SimpleTask.java).

Fixed this. Wasn't being used by the extensions library actually, it was just transitively using the library that contained the class.

**Comment by M66B** [comment]
You have requested a review, but I believe I have reviewed everything already?

**Comment by TacoTheDank** [comment]
> You have requested a review, but I believe I have reviewed everything already?

I have fixed the unresolved import error (the lifecycle service). Please review the commits and make sure you like them, thanks! :) (not meaning for this to sound passive-aggressive or anything lol)

**Pull request closed without merging**